exo-go/src/config/app_config_helpers.go

```diff
@@ -6,42 +6,40 @@ import (
 	"path"
 
 	"github.com/Originate/exosphere/exo-go/src/types"
-	"github.com/Originate/exosphere/exo-go/src/util"
 	"github.com/pkg/errors"
-	"github.com/segmentio/go-prompt"
+	prompt "github.com/segmentio/go-prompt"
 	"gopkg.in/yaml.v2"
 )
 
-// GetAllDependencyNames returns the container names (name+version) of all application
-// and service dependencies
-func GetAllDependencyNames(appDir string, appConfig types.AppConfig) ([]string, error) {
-	result := []string{}
-	for _, dependency := range appConfig.Dependencies {
-		containerNames := dependency.Name + dependency.Version
-		if !util.DoesStringArrayContain(result, containerNames) {
-			result = append(result, containerNames)
-		}
-	}
-	serviceConfigs, err := GetServiceConfigs(appDir, appConfig)
-	if err != nil {
-		return result, err
-	}
+// GetAllBuiltDependencies returns the AppDependency objects for the
+// dependencies of the entire application
+func GetAllBuiltDependencies(appConfig types.AppConfig, serviceConfigs map[string]types.ServiceConfig, appDir, homeDir string) map[string]AppDependency {
+	result := GetAppBuiltDependencies(appConfig, appDir, homeDir)
 	for _, serviceConfig := range serviceConfigs {
-		for _, containerName := range GetServiceDependencies(serviceConfig, appConfig) {
-			if !util.DoesStringArrayContain(result, containerName) {
-				result = append(result, containerName)
-			}
+		for dependencyName, builtDependency := range GetServiceBuiltDependencies(serviceConfig, appConfig, appDir, homeDir) {
+			result[dependencyName] = builtDependency
 		}
 	}
-	return result, nil
+	return result
+}
+
+// GetAppBuiltDependencies returns the AppDependency objects for the
+// dependencies defined in the given appConfig
+func GetAppBuiltDependencies(appConfig types.AppConfig, appDir, homeDir string) map[string]AppDependency {
+	result := map[string]AppDependency{}
+	for _, dependency := range appConfig.Dependencies {
+		builtDependency := NewAppDependency(dependency, appConfig, appDir, homeDir)
+		result[dependency.Name] = builtDependency
+	}
+	return result
 }
 
 // GetEnvironmentVariables returns the environment variables of
 // all dependencies listed in appConfig
 func GetEnvironmentVariables(appConfig types.AppConfig, appDir, homeDir string) map[string]string {
```

Comment:
Based on its usage I think GetEnvironmentVariables should take in the appBuiltDependencies instead of building them again.